02_bubble/jacklee/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(origin []int) []int {
+	result := make([]int, len(origin))
+	copy(result, origin)
+	arrLen := len(origin)
+	for i := 0; i < arrLen-1; i++ {
+		for j := 0; j < arrLen-1-i; j++ {
```

Comment:
If `i` counts down, you can simplify `j`'s loop. https://github.com/anz-bank/go-training/pull/114#discussion_r261849435